src/yaml/aslYamlLanguageService.ts

```diff
@@ -30,8 +31,40 @@ import {
 import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
 import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
 import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import { YAMLDocDiagnostic } from 'yaml-language-server/out/server/src/languageservice/utils/parseUtils'
 import doCompleteAsl from '../completion/completeAsl'
 
+function convertYAMLDiagnostic(yamlDiagnostic: YAMLDocDiagnostic, textDocument: TextDocument): Diagnostic {
+    const startLoc = yamlDiagnostic.location.start
+    let endLoc = yamlDiagnostic.location.end
+    let severity = yamlDiagnostic.severity
+
+    // Duplicate positining returns incorect end position and needs to be ovewritten
+    if (yamlDiagnostic.message === 'duplicate key') {
```

Comment:
Do we have a constants file with the diagnostics messages?

Reply:
I converted it to a constant.